go/marshal/decode.go

```diff
@@ -217,3 +224,31 @@ func nomsValueDecoder(v types.Value, rv reflect.Value) {
 	}
 	rv.Set(reflect.ValueOf(v))
 }
+
+func sliceDecoder(t reflect.Type) decoderFunc {
+	decoder := typeDecoder(t.Elem())
+	return func(v types.Value, rv reflect.Value) {
+		list := v.(types.List)
+		l := int(list.Len())
+		slice := reflect.MakeSlice(t, l, l)
```

Comment:
This creates a new slice. I'm not sure this is the desired behavior. The other places we just update the object passed in... I'll check what encoding/json does.